pkg/asserts/branch_exists.go

```diff
@@ -9,9 +9,8 @@ import (
 
 func BranchExists(t *testing.T, dir, expectedBranch string) {
 	t.Helper()
-	cmd := exec.Command("git", "branch")
+	cmd := exec.Command("git", "rev-parse", "--verify", "-q", "refs/heads/"+expectedBranch)
```

Comment:
To fix the linter warning:

```suggestion
	cmd := exec.Command("git", "rev-parse", "--verify", "-q", "refs/heads/"+expectedBranch) //nolint:gosec
```